### PR TITLE
Lavaland cafe

### PR DESCRIPTION
**Pull request by silicons**
image

Requested by @Wilchenx

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
Days without fetishist tricks: ~~1~~ 0

**Comment by kevinz000** [comment]
!(https://cdn.discordapp.com/attachments/317234835934674948/424169795911417856/unknown.png)

**Comment by 81Denton** [comment]
And here I was gonna add a kitchen to the seed vault and call it the lavaland seed vault & snack bar

**Comment by MMMiracles** [comment]
Don't you hate when you're at work setting questionably priced simple meals to hipsters when suddenly your workplace is transported to hell?

The clothing vendor in the back shows that the bistro is actually a front for the maid cafe so the space police wouldn't catch on.

**Comment by YakumoChen** [comment]
I don't feel like this is going to see use outside of memes. Space bar wasn't a rousing success either.

**Comment by Repukan** [comment]
Fetish content.

**Comment by Repukan** [comment]
You stop this shit or I'm going to mix cyanide into your catnip

**Comment by Bawhoppen** [comment]
Has far too much stuff in it. Not the most interesting mapping. No new original assets or features with it. 
Don't expect it to see much use in real rounds, either. 
Personally don't think we should add it.

**Comment by Wilchenx** [comment]
In the dark year of 2018 even cafes are considered a fetish content.

**Comment by JStheguy** [comment]
The only acceptable restaurants in the 26th century are dive bars and burger joints, anything else is heresy.

**Comment by Wilchenx** [comment]
Thing is, you can make it a FANCY burger joint if you want to. Nothing in this is predefined

Like a great coder once said

YOU CAN DO ANYTHING

**Comment by Shadowflame909** [comment]
I want a clown themed cafe or McDonald's! McDonald's! McDonald's!!!!

**Comment by Wilchenx** [comment]
We have food at home

**Comment by Tlaltecuhtli** [comment]
slippery slope

**Comment by Repukan** [comment]
*Drives into drive thru*
*Orders a black coffee, no sugar*
*Drive out*

**Comment by Shadowflame909** [comment]
You respond very fast

**Comment by Wilchenx** [comment]
@Tlaltecuhtli point where did you see slippery slope, we've been trying to catch him for years.

**Comment by myonn** [comment]
everything can and will be a fetish.

someone out there definitely has a thing for cafes transported to hellscapes, and I'm starting to connect the dots between that and a possible insecurity in repukan

**Comment by Repukan** [comment]
Let's close a PR for being fetish material and then reopen it but still add a way for whatever made it a fetish thing in the first place to still be accessible so I can put my dumb fetish into the game but uh oh bucko I didn't really make them spawn as maids so if they get the uniforms from the conveniently placed custome vendor it ain't me on yeah :^)

**Comment by Wilchenx** [comment]
You can PR removal of costume vendors to eradicate all fetishes alltogether if that really bothers you.

**Comment by Shadowflame909** [comment]
Cat girl ears with a maid costume. I uh..... Well there's more then one
costume choice. I hope for the best.

**Comment by kevinz000** [comment]
yell at wilchen/shezza about design issues i'm just coding and pring this for them because they're a lazy cat

**Comment by arsserpentarium** [comment]
Where did you find fetish content?
I didn't.

**Comment by arsserpentarium** [comment]
If you are talking about maids and catgirls, maids is needed profession in cafe.
Ca/tg/irls is part of TG culture. I don't like them very much myself, but it's part of local colorite and tradition.

**Comment by GuyonBroadway** [comment]
Why does a cafe have a botanical garden, protolathe and engineering vendors?

No other lavaland ruin gives you that much stuff besides free golems and they have to harvest thier own stuff to do anything.

Also ca/tg/ays are only a nu-tg thing.

**Comment by Wilchenx** [comment]
@GuyonBroadway because if something gets fucked you can then replace it using materials from botany, you are not supposed to leave the ruin anyway so it doesn't really matter.

**Comment by GuyonBroadway** [comment]
Yeah but all they need is a toolbox and a few sheets of metal, glass and wood. The beach bums don't get this much stuff, the animal hospital doesnt get this much stuff, the space hotel doesn't get this much stuff.  Heck the space hotel doesn't even get thier own hydroponics bay and considering they are more likely to get visitors that just goes to show how much stuff is crammed into one single lavaland outpost.

**Comment by Tlaltecuhtli** [comment]
whole point of those spawns was to learn specific jobs so they required adv tech shit

**Comment by Wilchenx** [comment]
Yeah the whole point of this is to mess around with food and have fun.

**Comment by AlexTheSysop42** [comment]
“Curiously enough, the only thing that went through the mind of the bowl of petunias as it fell was Oh no, not again. Many people have speculated that if we knew exactly why the bowl of petunias had thought that we would know a lot more about the nature of the Universe than we do now.”

**Comment by praisenarsie** [comment]
this is just powercreep ad infinitium to a shitty ruin that people will only play for like 2 days and no one is going to be down there anyway unless miners place a teleporter.

in fact it's a shitty reskin of the abandoned station & derelict

**Comment by praisenarsie** [comment]
> Thing is, you can make it a FANCY burger joint if you want to. Nothing in this is predefined
>
>Like a great coder once said
>
>YOU CAN DO ANYTHING

then what's wrong with spending like 15 minutes renovating the bar on the station into a cafe.

**Comment by Wilchenx** [comment]
>in fact it's a shitty reskin of the abandoned station & derelict

[citation needed]

**Comment by KorPhaeron** [comment]
I had vague plans to make a Restaurant At The End Of The World that was a space ship that landed in apocalyptic areas and had some ghost role patrons of various backgrounds/bartender/a way to let the mining shuttle land there and you should do that instead because Hitchhikers Guide is better than maid cafes

**Comment by KorPhaeron** [comment]
Also it provides a reason why there is a fucking resturaunt in the hellscape

**Comment by Wilchenx** [comment]
@KorPhaeron we never questioned beach biodome

**Comment by Shadowflame909** [comment]
These guys should be trained in the art of plasma fist. So they can defend
against the eventual ash lizard vs catgirl struggle. Or when the ash
lizards bring bubblegum to the cafe.

**Comment by Wilchenx** [comment]
@Shadowflame909 or they can powergame botany or die trying.

**Comment by Qustinnus** [comment]
@kevinz000 lol isn't this the type of shit you condemn goof for? "shit PR A was denied so I'll just make shit PR B which is really shit PR A in disguise"

**Comment by Wilchenx** [comment]
@Qustinnus i've talked to the maintainer that closed the PR
image
In case you've only read the title and saw the opportunity to shitpost i'm reposting the crucial information why this pr is open.

**Comment by Shadowflame909** [comment]
This is still catgirl maids though. Just with more disguises. If erp gets
enabled this rune will be a leaf hotspot for it. Honk

**Comment by Wilchenx** [comment]
They were never catgirls in the first place.

**Comment by Shadowflame909** [comment]
Aren't there cat ears in the vendor?

**Comment by Wilchenx** [comment]
There's also a cow in the pen.

**Comment by Shadowflame909** [comment]
If this does get merged first thing I'm doing is stealing the bananium tiles from the clown planet rune and turning this cafe into a clown themed one. Another idea is having an arcade machine in the cafe and taking the toy space suits and the space tiles and having a space themed version. add an arcade kevinz. Do it.

**Comment by Qustinnus** [comment]
@Wilchenx I read the PR and it only added the extra step of making you walk to a vendor

**Comment by kevinz000** [comment]
honestly kor's idea sounds better but i promised wilchen so whatever i'll code that after regardless

**Comment by sawrge** [comment]
if if's and but's were big fat nuts i'd have popped about a hundred

File: _maps/RandomRuins/LavaRuins/lavaland_surface_lavaland_cafe.dmm [hunk 1]
```diff
@@ -0,0 +1,1577 @@
...
+/turf/open/floor/plasteel/cafeteria,
+/area/ruin/powered/cafe)
+"aX" = (
+/turf/open/floor/plasteel/cafeteria,
+/area/ruin/powered/cafe)
+"aY" = (
+/obj/machinery/vending/wallmed{
+	name = "Emergency NanoMed";
+	pixel_y = 28;
+	req_access_txt = "0";
```
Comment (ShizCalev) [inline _maps/RandomRuins/LavaRuins/lavaland_surface_lavaland_cafe.dmm]: dv

File: _maps/RandomRuins/LavaRuins/lavaland_surface_lavaland_cafe.dmm [hunk 2]
```diff
@@ -0,0 +1,1577 @@
...
+"aA" = (
+/obj/machinery/light{
+	dir = 8
+	},
+/turf/open/floor/wood,
+/area/ruin/powered/cafe)
+"aB" = (
+/obj/structure/window/reinforced{
+	dir = 1;
+	pixel_y = 0
```
Comment (ShizCalev) [inline _maps/RandomRuins/LavaRuins/lavaland_surface_lavaland_cafe.dmm]: dv

File: _maps/RandomRuins/LavaRuins/lavaland_surface_lavaland_cafe.dmm [hunk 3]
```diff
@@ -0,0 +1,1577 @@
...
+	},
+/obj/machinery/door/window/eastright{
+	dir = 8;
+	name = "Animal Pen";
+	req_access_txt = "0"
```
Comment (ShizCalev) [inline _maps/RandomRuins/LavaRuins/lavaland_surface_lavaland_cafe.dmm]: dv

File: _maps/RandomRuins/LavaRuins/lavaland_surface_lavaland_cafe.dmm [hunk 4]
```diff
@@ -0,0 +1,1577 @@
...
+	},
+/turf/open/floor/grass,
+/area/ruin/powered/cafe)
+"aC" = (
+/obj/structure/window/reinforced{
+	dir = 1;
+	pixel_y = 0
```
Comment (ShizCalev) [inline _maps/RandomRuins/LavaRuins/lavaland_surface_lavaland_cafe.dmm]: dv

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by improvedname** [comment]
What worries me the whole structure is made out of wood and paper on a lava filled plaaanet

**Comment by GuyonBroadway** [comment]
Careful now, if you use logic you might upset the weeaboos and thier glorious folded a thousand times nippon paper houses.

**Comment by kevinz000** [comment]
@Wilchenx fix your map

**Comment by sawrge** [comment]
>tfw you walk into burger king and pop a half chub because your fetish is restaurants

**Comment by Qustinnus** [comment]
Knowing you you probably spotted the ballpit full of kids tbf @sawrge

**Comment by Bawhoppen** [comment]
Keep drama out. Github is only for serious business.

**Comment by ExcessiveUseOfCobblestone** [comment]
@Qustinnus not illegal when he's 15 tbf

**Comment by 81Denton** [comment]
- 4 (four) autodrobe resupply canisters so you can get kitty ears for the whole metagang BUT
- no shaker or bartending book
- no universal enzyme or a cookbook, no oil for the deep frier, no soap so the microwave doesn't explode
- no spare candles
- an autolathe but no materials to feed it with
- engivend has req_access_txt "11", meaning staff can't use it (or even hack it due to lack of multitool)
- a single beaker for all of botany, with no available mutagen to give chefs exotic ingredients
- wooden walls and paper windows, yet no wood for repairs or anything atmos related to repressurize, so staff and patrons don't suffocate when something punches a hole in the wall

cum on wilch you can do better

**Comment by AlexTheSysop42** [comment]
>4 (four) autodrobe resupply canisters so you can get kitty ears for the whole metagang

Maid outfits too, don't forget that

>no shaker

Might be in the bar vendor, not sure though.

>an autolathe but no materials to feed it with
engivend has req_access_txt "11", meaning staff can't use it (or even hack it due to lack of multitool)

You can probably use some spare tools from the vendor for a multitool and a bit of mats from the autolathe, but you still shouldn't have to scrounge up mats in what's supposed to be a well-stocked cafe.

>a single beaker for all of botany, with no available mutagen to give chefs exotic ingredients

Bottles can be made from the biogen and a competent botanist can use the chlorine gene from the blumpkin seeds with glowshrooms to get mutagen. It's pretty low-effort to just add some blumpkin seeds though, and shows that the botany area was added as an afterthought.

>no universal enzyme or a cookbook, no oil for the deep frier, no soap so the microwave doesn't explode

That makes the cafe part fairly useless, and clearly shows that it wasn't the main thing the author was thinking of. 

In general, nothing seems to be planned out very well except for the autodrobe restockers. This is definitely thinly-veiled fetish bait, and removing the starting maid costumes doesn't matter when someone can just get one from a vending machine.

**Comment by Wilchenx** [comment]
@81Denton this was made before cooking oil, added some
botany has blumpkins seeds from get go so mutagen is given for very little work
autolathe has materials now
engivend and other vendors removed because there's autolathe with materials
added one more beaker to botany, more can be made in autolathe
wood can be grown, steel can be grown too
made some atmos system that steals air from lavaland
spare candles added

**Comment by AlexTheSysop42** [comment]
You still haven't addressed the restocking units or the universal enzyme

**Comment by Wilchenx** [comment]
@AlexTheSysop42 restock canisters were added so players can decide on a gimmick if they want to and so everyone can follow it.

**Comment by Qustinnus** [comment]
mate i'm 100% sure if I open this map I'll find catears everywhere.

**Comment by Wilchenx** [comment]
@Qustinnus wanna bet 5$?

**Comment by AlexTheSysop42** [comment]
>so players can decide on a gimmick if they want to and so everyone can follow it

like catmaids

Also, that seems fairly thoroughly thought out compared to the lack of enzyme, only proving my point.

**Comment by arsserpentarium** [comment]
For guys, who are angry about maid outfit:did you know, that maid outfit isn't fetish content at first place,but formal uniform, used by workers of certain profession.
Guess, what is this profession and where do they work.

**Comment by Wilchenx** [comment]
@AlexTheSysop42 
image
here's your enzyme, i gathered some while you were salivating

**Comment by AlexTheSysop42** [comment]
Yes, people do use it to work, but it's also fetish content; that was established in the original PR and the "lack" of it is why this one is under consideration

**Comment by Wilchenx** [comment]
Now please stop trying to cause shitstorm and let me finish pipes so we can update the map

**Comment by AlexTheSysop42** [comment]
I'm making such a big deal over the enzyme because it's one of the key ingredients in almost anything a kitchen would make, and you _somehow_ forgot about that.

**Comment by Wilchenx** [comment]
I forgot to add it to the comment my friend, i never forgot about it.

**Comment by Wilchenx** [comment]
Well i did sort of.

**Comment by arsserpentarium** [comment]
If you aren't afraid for your sanity, give me any map and I'll find there fetish content.

**Comment by ExcessiveUseOfCobblestone** [comment]
2 showers for a 1-person bathroom certainly raises the question....

**Comment by Wilchenx** [comment]
@ExcessiveUseOfCobblestone people shower in clothes anyway

**Comment by AlexTheSysop42** [comment]
Even ignoring the fetish content, no one's even going to use the cafe. Just look at the beach bum ruin, how often does someone become a beach bum, stay in the biodome, and not commit suicide less than 5 minutes afterwards? In addition, it doesn't fit the theme of lavaland at all and (when you count the hydroponics area and tool vendors/lathe) doesn't have as much specialization as it should.

**Comment by Wilchenx** [comment]
Ruins that leave players to their own creativity like free golems and seed vault are often used to make fun things, I hope this one will provide similiar source of fun for players.

**Comment by Wilchenx** [comment]
The types of players that I expect to have fun from this ruin are mainly service departament players, so I added every service departament to the ruin.

**Comment by Wilchenx** [comment]
Well except janitors, because janitors are fetish content.

**Comment by AlexTheSysop42** [comment]
I'd be a lot more fine with this if the autodrobe had cat ears/maid outfits VV'd out and the chef had some kind of ingredient orderer (this _is_ a cafe, cafes don't normally grow ingredients)

**Comment by Wilchenx** [comment]
I don't think I need to cater to you if you're not the kind of player that would use this ruin.

**Comment by AlexTheSysop42** [comment]
>janitors are fetish content

what

**Comment by Wilchenx** [comment]
image

**Comment by ExcessiveUseOfCobblestone** [comment]
The kind of players that use this ruin..

Like space bar ruin...
Space hotel ruin...
Beach bar ruin?

**Comment by AlexTheSysop42** [comment]
>you're not the kind of player that would use this ruin

I'm trying to keep a half-baked, potentially unbalanced/incomplete ruin from being added to the game and crowding out other ruins.

**Comment by Wilchenx** [comment]
>I'm trying to keep a half-baked

Don't worry there will be plenty of baking!

**Comment by Shadowflame909** [comment]
Kek this rune already exists/existed 3 other times. I'm not sure cat ears
are enough to keep it from disappearing again. But a mime sleeper outta
work. Fake wall memes.

On Mar 20, 2018 11:13 AM, "Wilchenx" <notifications@github.com> wrote:

I'm trying to keep a half-baked

Don't worry there will be plenty of baking!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/tgstation/tgstation/pull/36519#issuecomment-374702416>,
or mute the thread

**Comment by 81Denton** [comment]
@Wilchenx I'd probably divide the ruin into multiple areas with their own air alarms so the place can repressurize properly.

**Review by ShizCalev — APPROVED** [review]

**Comment by ninjanomnom** [comment]
This just isn't interesting enough or worth adding to lavaland, cluttered as it is. No one on the team seems interested in it either.

**Comment by Iamgoofball** [comment]
@ExcessiveUseOfCobblestone it is at a federal and most state level

**Pull request closed without merging**